var/spack/repos/builtin/packages/qt/package.py

```diff
@@ -52,6 +44,34 @@ class Qt(Package):
     depends_on("libxcb")
 
 
+    def url_for_version(self, version):
+        url = "http://download.qt.io/archive/qt/"
+
+        if version >= Version('5'):
+            url += "%s/%s/single/qt-everywhere-opensource-src-%s.tar.gz" % \
+                    (version.up_to(2), version, version)
+        elif version >= Version('4.8'):
+            url += "%s/%s/qt-everywhere-opensource-src-%s.tar.gz" % \
+                    (version.up_to(2), version, version)
+        elif version >= Version('4.6'):
+            url += "%s/qt-everywhere-opensource-src-%s.tar.gz" % \
+                    (version.up_to(2), version)
+        elif version >= Version('4.0'):
+            url += "%s/qt-x11-opensource-src-%s.tar.gz" % \
+                    (version.up_to(2), version)
+        elif version >= Version('3'):
+            url += "%s/qt-x11-free-%s.tar.gz" % \
+                    (version.up_to(1), version)
+        elif version >= Version('2.1'):
```

Comment:
Qt 2 has probably been dead for a long time; no need to waste bytes to any support for it. [Here's](https://en.wikipedia.org/wiki/List_of_Qt_releases) the release history.